src/main/java/seedu/resireg/ui/PiechartPanel.java

```diff
@@ -0,0 +1,63 @@
+package seedu.resireg.ui;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.logging.Logger;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.chart.PieChart;
+import javafx.scene.layout.Region;
+import seedu.resireg.commons.core.LogsCenter;
+import seedu.resireg.model.allocation.Allocation;
+import seedu.resireg.model.room.Room;
+import seedu.resireg.model.student.Student;
+
+
+/**
+ * Panel containing the pie chart of allocated rooms.
+ */
+public class PiechartPanel extends UiPart<Region> {
+    private static final String FXML = "PiechartPanel.fxml";
+    private final Logger logger = LogsCenter.getLogger(PiechartPanel.class);
+
+    private final ObservableList<Student> studentList;
+    private final ObservableList<Allocation> allocationList;
+    private final ObservableList<Room> roomList;
+    private ObservableList<PieChart.Data> data;
+    @FXML
+    private PieChart piechart;
+
+    /**
+     * Creates a {@code PieChart} with the given {@code ObservableList}.
+     */
+    public PiechartPanel(ObservableList<Student> studentList,
+                         ObservableList<Allocation> allocationList,
+                         ObservableList<Room> roomList) {
+        super(FXML);
+        requireNonNull(studentList);
+        requireNonNull(allocationList);
+        requireNonNull(roomList);
+
+        this.studentList = studentList;
+        this.allocationList = allocationList;
+        this.roomList = roomList;
+
+        initData();
+        setChart();
+    }
+
+    private void initData() {
+        int allocatedCount = allocationList.size();
+        int unallocatedCount = roomList.size() - allocatedCount;
```

Comment:
might want to add an assertion here, that `unallocatedCount > 0`

Reply:
I think that unallocatedCount can be 0.

Reply:
Good point!

Reply:
Added assertion for `unallocatedCount >= 0`.